### PR TITLE
TLPORTAL-383: fix regression with ctools assignments feed

### PR DESCRIPTION
**Pull request by dlhaines**

### CODE REVIEW

File: server/ctools_direct_response.rb [hunk 1]
```diff
@@ -91,14 +91,24 @@ def self.filter(assignment)
...
+
+    if 'assignment.due.date'.casecmp(calendarTimeLabelKey) != 0
+      logger.debug "#{self.class.to_s}:#{__method__}:#{__LINE__}: assignment: fail for bad key: [#{calendarTimeLabelKey}]"
+      return nil
+    end
 
-    return nil unless calendarTimeLabelKey == 'assignment.due.date'
-    
-    return assignment
+    assignment
```
Comment (zqian) [inline server/ctools_direct_response.rb]: should this be "return assignment" instead?

File: server/ctools_direct_response.rb [hunk 2]
```diff
@@ -91,14 +91,24 @@ def self.filter(assignment)
 
     logger.debug "#{self.class.to_s}:#{__method__}:#{__LINE__}: assignment: [#{assignment.inspect}]"
     calendarItem = assignment['calendarItem']
-    return nil if calendarItem.nil?
+
+    if calendarItem.nil?
+      logger.debug "#{self.class.to_s}:#{__method__}:#{__LINE__}: assignment: fail for nil calendarItem"
```
Comment (zqian) [inline server/ctools_direct_response.rb]: How about log the assignment id also for future reference?

File: server/ctools_direct_response.rb (hunk 1 above)
Comment (dlhaines) [inline server/ctools_direct_response.rb]: The value of the late expression in a method is used as the return value even without an explicit return statement.  I have been using this convention for consistency.

File: server/ctools_direct_response.rb (hunk 2 above)
Comment (dlhaines) [inline server/ctools_direct_response.rb]: The entire assignment is logged in an earlier debug statement, so it doesn't need to be repeated here.  That would be a good idea if/when the earlier log statement is taken out.

**Pull request merged**